### DISCUSSION TITLE
Waydroid inside Android

### DISCUSSION THREAD

**Discussion by xdbgjhjreedwef** [waydroid/waydroid#1301] [Ideas]
Today there are lots of scary apps (social, banking, etc), which must be isolated from the main os. It would be convenient, if we could use such apps inside a container on Android. Can we run Waydroid inside Android ?

**Comment by Runa-Chin** [comment]
I think it still impossible these times

**Reply by korpe1** [reply]
Why?

**Reply by Runa-Chin** [reply]
As you know, Waydroid itself are Python-based program and Android app couldn't run that directly. But if you ask on Termux, it's still imposible because Waydroid need to communicate with kernel module `binder` and `ashmem` for run the Waydroid images. Meanwhile, Android itself are running under customized Linux kernel and Android on our device (smartphone/tablet/automotive) had `binder` module running.

And Waydroid are not designed to run inside Android or "Android runs another Android". Did you remember Anbox ? The precursor of Waydroid also not designed for run inside Android or what I'd said "Android runs another Android". That's why I say impossible.

Even could with Termux, are Termux itself could communicate with kernel ? Termux is just subsystems, even on rooted Android.

**Comment by Runa-Chin** [comment]
> Today there are lots of scary apps (social, banking, etc), which must be isolated from the main os. It would be convenient, if we could use such apps inside a container on Android. Can we run Waydroid inside Android ?

I guess the answer for this are "Making Work profile on your smartphone using Shelter app" instead forcing to run Waydroid which is imposible to "Android runs another Android". Even could with another app like VMOS, it's still unstable for severe phone.
